File: Data Structures & Algorithms/is-anagram/submission-0.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        seen = {}

        for c in s:
            if c not in seen:
                seen[c] = 1
            else:
                seen[c] += 1

        for char in t:
            if char not in seen:
                return False
            seen[char] -= 1

        for val in seen.values():
            if val != 0:
                return False

        return True

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        set<int> seen;

        for (int num : nums) {
            if (seen.find(num) != seen.end()) {
                return num;
            }
            else {
                seen.insert(num);
            }
        }
    }
};
